src/services/earn/earn-service.ts

```diff
@@ -137,6 +144,109 @@ export class EarnService implements IEarnService {
     // Build multicall and return tx
     return buildCompanionMulticall({ chainId, calls, value: depositInfo.value });
   }
+
+  async buildIncreasePositionTx({ chainId, positionId, increase, permissionPermit }: IncreaseEarnPositionParams): Promise<BuiltTransaction> {
+    const vault = EARN_VAULT.address(chainId);
+    const companion = EARN_VAULT_COMPANION.address(chainId);
+    let increaseInfo: { token: Address; amount: bigint; value: bigint };
+    if (!increase) {
+      increaseInfo = { token: Addresses.ZERO_ADDRESS, amount: 0n, value: 0n };
+    } else if ('token' in increase) {
+      const amount = BigInt(increase.amount);
+      increaseInfo = { token: increase.token, amount, value: isSameAddress(increase.token, Addresses.NATIVE_TOKEN) ? amount : 0n };
+    } else {
+      increaseInfo = { token: increase.permitData.token, amount: BigInt(increase.permitData.amount), value: 0n };
+    }
+
+    const bigIntPositionId = BigInt(positionId);
+    const [positionOwner, { needsSwap, asset }] = await Promise.all([
+      this.providerService.getViemPublicClient({ chainId }).readContract({
+        abi: vaultAbi,
+        address: vault,
+        functionName: 'ownerOf',
+        args: [bigIntPositionId],
+      }),
+      this.checkIfNeedsSwap({ chainId, strategyId: bigIntPositionId, depositToken: increaseInfo.token }),
+    ]);
+    const callVaultDirectly = !increase || increaseInfo.amount === 0n || ('token' in increase && !needsSwap);
```

Comment:
Same with increase not being defined or amount being zero. Actually, if amount is zero let's throw an error, since the contract would fail when called